src/NServiceBus.Persistence.AzureStorage/SagaPersisters/AzureSagaPersister.cs

```diff
@@ -56,7 +56,11 @@ public Task Update(IContainSagaData sagaData, SynchronizedStorageSession session
                 EntityProperty value;
                 if (tableEntity.TryGetValue(SecondaryIndexIndicatorProperty, out value))
                 {
-                    meta.AddSecondaryIndexId(entity, PartitionRowKeyTuple.Parse(value.StringValue));
+                    var partitionRowKeyTuple = PartitionRowKeyTuple.Parse(value.StringValue);
+                    if (partitionRowKeyTuple.HasValue)
```

Comment:
Is this fixing a bug? Not sure why this PR is in place other than handle the case that `partitionRowKeyTuple` is null and we shouldn't use it... @danielmarbach ?